triggers/triggers-dt/src/main/java/gg/xp/xivsupport/triggers/Arcadion/M4N.java

```diff
@@ -0,0 +1,126 @@
+import gg.xp.xivsupport.events.actlines.events.vfx.StatusLoopVfxApplied;
+import gg.xp.xivsupport.events.state.XivState;
+import gg.xp.xivsupport.events.triggers.seq.SequentialTrigger;
+import gg.xp.xivsupport.events.triggers.seq.SqtTemplates;
+import gg.xp.xivsupport.events.triggers.support.NpcAbilityUsedCallout;
+import gg.xp.xivsupport.events.triggers.support.NpcCastCallout;
+import gg.xp.xivsupport.models.ArenaPos;
+import gg.xp.xivsupport.models.ArenaSector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+@CalloutRepo(name = "M4N", duty = KnownDuty.M4N)
+public class M4N extends AutoChildEventHandler implements FilteredEventHandler {
+	public static final Logger log = LoggerFactory.getLogger(M4N.class);
+
+	@NpcCastCallout({0x92BD, 0x92BF}) //second ID is while wings are active
+	private final ModifiableCallout<AbilityCastStart> sidewiseSparkWest = new ModifiableCallout<>("Sidewise Spark: East safe", "East");
+	@NpcCastCallout({0x92BC, 0x92BE})
+	private final ModifiableCallout<AbilityCastStart> sidewiseSparkEast = new ModifiableCallout<>("Sidewise Spark: West safe", "West");
+	@NpcAbilityUsedCallout(0x92AB)
+	private final ModifiableCallout<AbilityUsedEvent> stampedingThunderWest = new ModifiableCallout<>("Stampeding Thunder: Go East", "Go East");
+	@NpcAbilityUsedCallout(0x92AC)
+	private final ModifiableCallout<AbilityUsedEvent> stampedingThunderEast = new ModifiableCallout<>("Stampeding Thunder: Go West", "Go West");
+
+	public M4N(XivState state) {
+		this.state = state;
+	}
+	private final XivState state;
+
+	@Override
+	public boolean enabled(EventContext context) {
+		return state.dutyIs(KnownDuty.M4N);
+	}
+
+	public static final List<Long> validStacks = List.of(723L,724L);
+
+	public ArenaSector getSafeDirectionForBA(BuffApplied ba) {
+		//723 hits north, go south
+		//724 hits south, go north
+		long stacks = ba.getRawStacks();
+		return stacks == 723 ? ArenaSector.SOUTH : ArenaSector.NORTH;
+	}
+
+	private final ModifiableCallout<?> gunStart = new ModifiableCallout<>("Gun Blasts: Start", "Start {dir}");
+	private final ModifiableCallout<?> gunSafe = new ModifiableCallout<>("Gun Blasts: Next Safe", "{dir}");
+
+	@AutoFeed
+	private final SequentialTrigger<BaseEvent> gunBlast = SqtTemplates.sq(30_000, AbilityCastStart.class,
+			acs -> acs.abilityIdMatches(0x92B0, 0x9B4F, 0x9B56, 0x92AD, 0x9B55, 0x9B57),
+			(e1, s) -> {
+				log.info("Gun Blasts: Start");
+				int iterations = switch((int)e1.getAbility().getId()) {
+					case 0x92B0, 0x92AD -> 2;
+					case 0x9B4F, 0x9B55 -> 3;
+					case 0x9B56, 0x9B57 -> 4;
+					default -> 0;
+				};
+
+				//Buff 0xB9A
+				ArenaSector buff1 = getSafeDirectionForBA(s.waitEvent(BuffApplied.class, ba -> validStacks.contains(ba.getRawStacks()) && ba.buffIdMatches(0xB9A)));
+				log.info("Gun Blasts Start: {}", buff1);
+				s.updateCall(gunStart.getModified(Map.of("dir", buff1)));
```

Comment:
You can use the shorter updateCall syntax (`s.updateCall(gunStart)`) by using `s.setParam(key, value)` before to set the parameters.